### DISCUSSION TITLE
Removing a layer - UI update?

### DISCUSSION THREAD

**Discussion by jameslavery** [Mapsui/Mapsui#2218] [Q&A] [answered]
I want to allow the user to add/remove/clear layers at runtime. 

The only layer I've currently got in the collection is a raster layer with a GeoTIFF in it.

I'm making a call to Map.Layers.Clear(), but the UI isn't updating. Is there something else I need to do? Is there a restriction on removing all layers?

**Comment by pauldendulk** [comment, marked as answer]
You need to call `Map.RefreshGraphics();` After that the map will be an empty white surface.

**Reply by jameslavery** [reply]
Brilliant - thanks. Is that documented anywhere? I did look, but couldn't find 'general lifecycle' info.

**Reply by pauldendulk** [reply]
btw, you can still pan and zoom on the invisible map, this may have an unexpected effect if data is added again. You may want to PanLock/ZoomLock the screen, or you may want to zoom to an extent if you add a layer again.

**Reply by jameslavery** [reply]
Thanks - good point. We're generally going to have at least one layer (this is just my initial 'finding my feet' code.

**Reply by pauldendulk** [reply]
About documentation. I don't think we have good documentation about this. We have this piece about asynchronous data fetching, but it is also may be outdated. https://mapsui.com/documentation/async-fetching.html. Here is a bit about the core components, which should be improved and get a more prominent place in the docs https://mapsui.com/documentation/mapsui-components.html. In this case it is more about how the render loop works. It would be good to write something about it. The general idea of Mapsui is that it consists of lego blocks that can be understood by developers and combined to their needs, as opposed to software where you only need to write one single line and then everything magically works. In order to let developers use it like that they should know a bit more about it.
